examples/material/create_material.py

```diff
@@ -1,28 +1,58 @@
 #!/usr/bin/env python
 # coding: utf-8
 
+# [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Exabyte-io/exabyte-api-examples/blob/feature/SOF-4400-skinny-req/examples/material/create_material.ipynb)
+
 # # Overview
 # 
 # In this example we create a material from a JSON config with [tags](https://docs.exabyte.io/entities-general/data/#tags) to identify the material.
 
-# # Execution
-# 
-# > <span style="color: orange">**NOTE**</span>: In order to run this example, an active Exabyte.io account is required. RESTful API credentials shall be updated in [settings](../settings.py). The generation of the credentials is also explained therein.
+# # Complete Authorization Form and Initialize Settings
 # 
-# ## Import packages
+# This will also determine environment and set all environment variables. We determine if we are using Jupyter Notebooks or Google Colab to run this tutorial. If google drive, we will have some extra steps to do.
+
+# In[ ]:
+
+
+#@title Authorization Form
+ACCOUNT_ID = "" #@param {type:"string"}
+AUTH_TOKEN = "" #@param {type:"string"}
+MATERIALS_PROJECT_API_KEY = "" #@param {type:"string"}
+ORGANIZATION_ID = ""#@param {type:"string"}
 
-# In[]:
+import os, glob
 
+if 'google.colab' in str(get_ipython()):
+  from google.colab import _message
+  notebook_name = _message.blocking_request('get_ipynb')['ipynb']['metadata']['colab']['name']
+  get_ipython().system('git clone -b feature/SOF-4400-skinny-req https://github.com/Exabyte-io/exabyte-api-examples.git    ')
+  notebook_path = glob.glob('**/'+notebook_name, recursive=True)[0][0:-len(notebook_name)]
+  os.chdir(notebook_path)
+  get_ipython().system('pip install --no-deps -r ../../requirements-colab.txt')
+  os.environ['notebook_environment'] = "Colab"
+elif 'ZMQInteractiveShell'  in str(get_ipython()):
+  os.environ['notebook_environment'] = "Jupyter"
+else:
+  os.environ['notebook_environment'] = ""
 
-import os
-import sys
 
-# Import settings and utils file
+# # Imports
+
+# In[ ]:
+
+
+import sys, importlib
 module_path = os.path.abspath(os.path.join('..'))
 if module_path not in sys.path: sys.path.append(module_path)
+from utils.generic import display_JSON, update_settings_with_users_authorization_info
+update_settings_with_users_authorization_info(ACCOUNT_ID, AUTH_TOKEN, MATERIALS_PROJECT_API_KEY, ORGANIZATION_ID)
```

Comment:
Should this only be run for Colab?

Reply:
When we run this in Jupyter there is no need to update the settings file after the tokens are set once. So we can adjust the logic in the update function to do nothing if the values in the file are not changing

Reply:
For that, we can switch to using `settings.json` with something like:

```json
{
    AUTH_TOKEN: "value",
    "KEY": "value"
}
```

in settings.py we would just read those values and expose them in py runtime for convenience.

When we want to modify the values, we'll modify the JSON file instead of py